cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -221,6 +233,75 @@ class DMSRules(BaseRules):
+            if len(list_definitions) > 1:
+                exceptions.append(
+                    f"Container {container}.{prop_name} is defined with different "
+                    f"list definitions: {list_definitions}"
+                )
+            nullable_definitions = {prop.nullable for prop in properties if prop.nullable is not None}
+            if len(nullable_definitions) > 1:
+                exceptions.append(
+                    f"Container {container}.{prop_name} is defined with different "
+                    f"nullable definitions: {nullable_definitions}"
+                )
+            default_definitions = {prop.default for prop in properties if prop.default is not None}
+            if len(default_definitions) > 1:
+                exceptions.append(
+                    f"Container {container}.{prop_name} is defined with different "
+                    f"default definitions: {default_definitions}"
+                )
+            index_definitions = {",".join(prop.index) for prop in properties if prop.index is not None}
+            if len(index_definitions) > 1:
+                exceptions.append(
+                    f"Container {container}.{prop_name} is defined with different "
+                    f"index definitions: {index_definitions}"
+                )
+            constraint_definitions = {",".join(prop.constraint) for prop in properties if prop.constraint is not None}
+            if len(constraint_definitions) > 1:
+                exceptions.append(
+                    f"Container {container}.{prop_name} is defined with different "
+                    f"unique constraint definitions: {constraint_definitions}"
+                )
+
+            # This sets the container definition for all the properties where it is not defined.
+            # This allows the user to define the container only once.
+            value_type = value_types.pop()
+            list_definition = list_definitions.pop() if list_definitions else None
+            nullable_definition = nullable_definitions.pop() if nullable_definitions else None
+            default_definition = default_definitions.pop() if default_definitions else None
+            index_definition = index_definitions.pop().split(",") if index_definitions else None
+            constraint_definition = constraint_definitions.pop().split(",") if constraint_definitions else None
+            for prop in properties:
+                prop.value_type = value_type
+                prop.is_list = prop.is_list or list_definition
+                prop.nullable = prop.nullable or nullable_definition
+                prop.default = prop.default or default_definition
+                prop.index = prop.index or index_definition
+                prop.constraint = prop.constraint or constraint_definition
+
+        if exceptions:
+            exception_str = "\n".join(exceptions)
+            raise ValueError(f"Inconsistent container(s): {exception_str}")
```

Comment:
You can actually use `ExceptionGroup` for this, 

For python lower than 3.11, make sure to import this:

```
if sys.version_info < (3, 11):
    from exceptiongroup import ExceptionGroup
```

Reply:
I tried, but pydantic was not happy.